lib/extensions/firestore_extensions.dart

```diff
@@ -73,11 +79,15 @@ extension ConnectAndConvert on FirebaseFirestore {
                 Message(text: docSnapshot.data()['text'] as String))),
       ));
     }, onError: (dynamic error, StackTrace trace) {
-      controller.add(AddProblem.from(
-        message: error.toString(),
-        traceString: trace.toString(),
-        type: ProblemType.observeMessages,
-      ));
+      controller.add(
+        AddProblem(
+          problem: ObserveMessagesProblem.by(
```

Comment:
An "ObserveMessages" event is perhaps a 'larger' event, containing the "ConnectToMessages" event. I wonder if changing the name to be more specific would be helpful. What do you think? Just thinking out loud at this stage.

Reply:
Agree, that would be in keeping with naming the problem after the action being done.
Fixed!